util/src/main/scala/com/scalableminds/util/io/ZipIO.scala

```diff
@@ -100,11 +100,24 @@ object ZipIO {
 
     def isFileHidden(e: ZipEntry): Boolean = new File(e.getName).isHidden || e.getName.startsWith("__MACOSX")
 
+    def stripPathFrom(path: Path, string: String): Option[String] = {
+      if(path.toString.contains(string))
+        Some(path.toString.substring(0, path.toString.indexOf(string)))
+      else
+        None
+    }
+
     import collection.JavaConverters._
     val zipEntries = zip.entries.asScala.filter(e => !e.isDirectory && (includeHiddenFiles || !isFileHidden(e))).toList
 
+    //color, mask, segmentation
     val commonPrefix = if (truncateCommonPrefix) {
-      PathUtils.commonPrefix(zipEntries.map(e => Paths.get(e.getName)))
+      val commonPrefixNotFixed = PathUtils.commonPrefix(zipEntries.map(e => Paths.get(e.getName)))
+      val strippedPaths = List("color", "mask", "segmentation").flatMap(stripPathFrom(commonPrefixNotFixed, _))
```

Comment:
Why is it correct to strip either `color` or `mask` or `segmentation`? Why never all?

Reply:
Because a dataLayer only has one category it belongs to.
Norman just confirmed this

Reply:
Ok, perfect. Maybe add a comment about it then, it's not super obvious at first sight.